plugins/in_tail/tail_file.c

```diff
@@ -399,26 +399,33 @@ static void cb_results(const char *name, const char *value,
                        size_t vlen, void *data)
 {
     struct flb_hash *ht = data;
+    char *str;
     char *p;
 
-    while ((p = strchr(value, '.'))) {
+    if (vlen == 0) {
+        return;
+    }
+
+    str = flb_strndup(value, vlen);
```

Comment:
why the new memory allocation ?

Reply:
Because value is const and *p = '_' violates its constness. There is no compiler warning because strchr() returns a pointer to non const char I guess. It's interresting, I'll see the implementation of strchr ;-)

Reply:
It seems that strchr() silently discards constness https://opensource.apple.com/source/BerkeleyDB/BerkeleyDB-18/db/clib/strchr.c.auto.html.

Reply:
This should not be an issue, note that our hash implementation always creates a new memory entry for the key and the value. The extra allocation is not needed and is just adding overhead for every processed record.

Reply:
The problem doesn't happen on the copy inside the hash table: when doing `*p = '_';` it modifies path which is passed as parameter to tag_compose() and then to flb_regex_do() inside tag_compose().

Reply:
Anyway, we can just drop this loop as it has proven posing problems in certain circumstances.

Reply:
I dropped the loop (and thus the allocation). The escaping could be added back later at the line just after flb_hash_get() conditioned by an option.

Reply:
Seems like a legit concern to me.  We're modifying the value that was passed as `const` which might not be what the calling code is expecting.  If a is calling b and c with the parameter, but b is modifying that along the way, is c expecting that, or not?  Generally not.

Reply:
why do we need that transformation from __a.b__ to __a_b__ ?

Reply:
When you configure a tag of the form part-1.part-2.part-3 and for example part-2 contains a dot, the tag will finally look like part-1.part-2.part-3.part-4 which could mess your tag based routing after that. Escaping dots was a way to avoid this. But it introduced problems, for example with Kubernetes filter, when you have a pod which have a dot in its name (which is forbidden per the spec but not enforced for a reason I don't know), escaping this dot makes the filter don't find its metadata as it asks the K8s API server with the escaped name. I don't see a proper solution for this case, other than rewriting tag in the Kubernetes filter. For other cases, maybe having a configurable escaping could be good. For the moment, let's drop the escaping.